### PR TITLE
Use zero-based indexes when referring to option indexes.

### PR DESCRIPTION
**Pull request by dwsteele**
There were a number of places in the code where "hostId" was used, but hostId is just the option group index + 1 so this led to a lot of +1 and -1 to convert the id to an index and vice versa.

Instead just use the zero based index wherever possible. This is pretty much everywhere except when the host-id option is read or set, or where a message is being formatted for the user.

Also fix a bug in protocolRemoteParam() where remotes spawned from the main process could get process ids that were not 0. Only the locals should spawn remotes with process id > 0. This seems to have been harmless since the process id is only a label, but it could be confusing when debugging.

### CODE REVIEW

**Review by cmwshang — CHANGES_REQUESTED** [review]
Review complete

File: src/protocol/helper.c [hunk 1]
```diff
@@ -304,12 +295,12 @@ protocolRemoteParam(ProtocolStorageType protocolStorageType, unsigned int protoc
     strLstAddZ(result, "PasswordAuthentication=no");
 
     // Append port if specified
-    ConfigOption optHostPort = isRepo ? cfgOptRepoHostPort : cfgOptPgHostPort + hostIdx;
+    ConfigOption optHostPort = isRepo ? cfgOptRepoHostPort : cfgOptPgHostPort;
 
     if (cfgOptionTest(optHostPort))
```
[the remark this answers is not in the document]
Comment (dwsteele) [reply src/protocol/helper.c:300]: Yeah, this was an ill-advised merge from the next commit. Ironically I was trying to make things simpler. I'll just revert and we'll try again when it makes more sense.

File: src/protocol/helper.c [hunk 2]
```diff
@@ -323,21 +314,24 @@ protocolRemoteParam(ProtocolStorageType protocolStorageType, unsigned int protoc
     KeyValue *optionReplace = kvNew();
 
     // Replace config options with the host versions
-    unsigned int optConfig = isRepo ? cfgOptRepoHostConfig : cfgOptPgHostConfig + hostIdx;
+    unsigned int optConfig = isRepo ? cfgOptRepoHostConfig : cfgOptPgHostConfig;
```
[the remark this answers is not in the document]
Comment (dwsteele) [reply src/protocol/helper.c:317]: Same as above.

File: src/protocol/helper.c [hunk 3]
```diff
@@ -488,35 +481,33 @@ protocolRemoteGet(ProtocolStorageType protocolStorageType, unsigned int hostId)
...
         {
-            unsigned int optHost = isRepo ? cfgOptRepoHost : cfgOptPgHost + hostId - 1;
+            unsigned int optHost = isRepo ? cfgOptRepoHost : cfgOptPgHost;
+            unsigned int optHostIdx = isRepo ? 0 : hostIdx;
 
             // Execute the protocol command
             protocolHelperClient->exec = execNew(
-                cfgOptionStr(cfgOptCmdSsh), protocolRemoteParam(protocolStorageType, protocolId, hostId - 1),
-                strNewFmt(PROTOCOL_SERVICE_REMOTE "-%u process on '%s'", protocolId, strZ(cfgOptionStr(optHost))),
+                cfgOptionStr(cfgOptCmdSsh), protocolRemoteParam(protocolStorageType, hostIdx),
```
[the remark this answers is not in the document]
Comment (dwsteele) [reply src/protocol/helper.c:503]: At this point optHostIdx and hostIdx have become the same thing so I'll just replace optHostIdx with hostIdx.

File: src/command/check/common.c [hunk 4]
```diff
@@ -42,32 +42,31 @@ checkArchiveCommand(const String *archiveCommand)
...
     FUNCTION_TEST_END();
 
-    ASSERT(dbIdx > 0);
     ASSERT(dbObject != NULL);
 
     MEM_CONTEXT_TEMP_BEGIN()
     {
         unsigned int dbVersion = dbPgVersion(dbObject);
         const String *dbPath = dbPgDataPath(dbObject);
-        unsigned int pgPath = cfgOptPgPath + (dbIdx - 1);
```
[the remark this answers is not in the document]
Comment (dwsteele) [reply src/command/check/common.c:61]: Again, I was trying to make less churn with the next commit, but the changes don't make much sense in this context. I reverted and we'll revisit in the next commit.

**Review by cmwshang — APPROVED** [review]

**Pull request merged**